### PR TITLE
Condition output persistance

### PR DESCRIPTION
**Pull request by JoelMcKinnon**
Added a call to this.persist() in checkInputValue(). This function name was misleading, so changed it to setOutputValue(). Another function was called setOutput and I changed that one to setOutputSelection().

Resolves issue: #2690

### CODE REVIEW

**Comment by shefalijoshi** [comment]
@JoelMcKinnon Tests failing, please fix them and I'll approve the PR. Thanks.

File: src/plugins/condition/components/Condition.vue [hunk 1]
```diff
@@ -212,7 +212,8 @@ export default {
             domainObject: this.domainObject,
             currentCriteria: this.currentCriteria,
             expanded: true,
-            selectedOutputKey: '',
+            trigger: 'all',
+            selectedOutputSelection: '',
```
Comment (shefalijoshi) [inline src/plugins/condition/components/Condition.vue]: Is `stringOutputField` used anywhere? If not, please remove it.

File: src/plugins/condition/components/Condition.vue [hunk 2]
```diff
@@ -229,7 +230,29 @@ export default {
     },
     methods: {
         initialize() {
-            this.setOutput();
+            this.setOutputSelection();
+            if (!this.domainObject.isDefault) {
+                this.conditionClass = new ConditionClass(this.domainObject, this.openmct);
```
Comment (shefalijoshi) [inline src/plugins/condition/components/Condition.vue]: This code was removed in topic-conditionals. Can you double check that your code is upto date with topic-conditionals and you didn't accidentally add some stuff back?

File: src/plugins/condition/components/Condition.vue [hunk 3]
```diff
@@ -256,6 +279,9 @@ export default {
                 index: Number(ev.target.closest('.widget-condition').getAttribute('data-condition-index'))
             });
         },
+        updateCurrentCondition() {
```
Comment (shefalijoshi) [inline src/plugins/condition/components/Condition.vue]: We no longer need this either. I think it has been removed in the topic-conditionals branch.
Can you do a compare with topic-conditionals to make sure?

File: src/plugins/condition/components/Condition.vue (hunk 1 above)
Comment (JoelMcKinnon) [reply src/plugins/condition/components/Condition.vue]: done

File: src/plugins/condition/components/Condition.vue (hunk 2 above)
Comment (JoelMcKinnon) [reply src/plugins/condition/components/Condition.vue]: done

File: src/plugins/condition/components/Condition.vue (hunk 3 above)
Comment (JoelMcKinnon) [reply src/plugins/condition/components/Condition.vue]: Done

**Pull request closed without merging**